app/code/Magento/CatalogImportExport/Model/Import/Product.php

```diff
@@ -1737,7 +1749,43 @@ protected function _saveProducts()
                 $position = 0;
                 foreach ($rowImages as $column => $columnImages) {
                     foreach ($columnImages as $columnImageKey => $columnImage) {
-                        if (!isset($uploadedImages[$columnImage])) {
+                        $hash = '';
+                        if (filter_var($columnImage, FILTER_VALIDATE_URL) === false) {
+                            $filename = $importDir . DIRECTORY_SEPARATOR . $columnImage;
+                            if ($this->fileDriver->isExists($filename)) {
+                                $hash = hash_file(
+                                    'sha256',
+                                    $importDir . DIRECTORY_SEPARATOR . $columnImage
+                                );
+                            }
+                        } else {
+                            $hash = hash_file('sha256', $columnImage);
+                        }
+
+                        // Add new images
+                        if (empty($rowExistingImages)) {
```

Comment:
If we need a comment to explain the intention it may worth creating a private method with a descriptive name.

Reply:
I actually would recommend wrapping this `ifelse` within a function that returns a bool, and just use it on row 1786 instead of defining a variable.